pkg/runtime/runtime.go

```diff
@@ -583,6 +587,22 @@ func (a *DaprRuntime) appHealthReadyInit(opts *runtimeOpts) {
 		} else {
 			a.daprHTTPAPI.SetActorRuntime(a.actor)
 			a.daprGRPCAPI.SetActorRuntime(a.actor)
+
+			wfComponentFactory := wfengine.BuiltinWorkflowFactory(a.workflowEngine)
+
+			if wfInitErr := a.workflowEngine.SetActorRuntime(a.actor); wfInitErr != nil {
+				log.Warnf("Failed to set actor runtime for Dapr workflow engine - workflow engine will not start: %w", wfInitErr)
+			} else {
+				if a.workflowComponentRegistry != nil {
+					log.Infof("Registering component for dapr workflow engine...")
+					a.workflowComponentRegistry.RegisterComponent(wfComponentFactory, "dapr")
+					if componentInitErr := a.initWorkflowComponent(wfengine.ComponentDefinition); componentInitErr != nil {
+						log.Warnf("Failed to initialize Dapr workflow component: %v", componentInitErr)
+					}
+				} else {
+					log.Infof("No workflow registry available, not registering Dapr workflow component...")
+				}
+			}
```

Comment:
can we move this logic to something like `initWfEngine()` method ? I guess that it would improve readability

Reply:
Agreed. Done.